### ISSUE TITLE
Allow facilities to remove objects from locations

### ISSUE THREAD

**Issue by grizz** [peeringdb/peeringdb#1366]
Along with #1344 and using facility network policy for carriers, facility owners would like to be able to remove objects from being listed at their facilities.

This would include, IXes, Networks, and Carriers.

+1 from me

**Comment by mustangthz** [comment]
+1

**Comment by martinhannigan** [comment]
+1



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Matt Griswold ***@***.***>
Sent: Tuesday, April 4, 2023 7:53:22 PM
To: peeringdb/peeringdb ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [peeringdb/peeringdb] allow facilities to remove objects from locations (Issue #1366)


Along with #1344<https://github.com/peeringdb/peeringdb/issues/1344> and using facility network policy for carriers, facility owners would like to be able to remove objects from being listed at their facilities.

This would include, IXes, Networks, and Carriers.

+1 from me

**Comment by arnoldnipper** [comment]
+1

**Comment by mcmanuss8** [comment]
+1

**Comment by arnoldnipper** [comment]
We have a similar feature request from IX operators (see #1272). Does it make sense to combine both issues? At least for the GUI, it should. Under the hood, #1271 prolly is implemented via the existing DOTF procedures

**Comment by arnoldnipper** [comment]
The network IMO only can, if the facility is allowing subleasing. Hence, it would also allow the sub-lessee to be shown.

**Comment by martinhannigan** [comment]
This scenario is something that is already contractually enforceable. To be
pure as to who is actually “in” a facility, not allowing what is
effectively type 2 is certainly an option.  A large $DC (or three) actually
enforce “naughty” lists. The benefit for us is knowing who is physically
there, which IXP is really remote and who is type 2 IMHO. Did I miss/get
that right?

If we intend to focus on address in v3, this is much more applicable to
that. An address should list fiber operators and “facilities” within an
address should list who’s there. If they don’t want to? Self inflicted
gunshot wound IMHO.

I don’t see any antitrust risk here. As long as the facility treats the
connections equal. But that liability is theirs. We’re just providing the
mechanism.

Hope that helps!

(American) Marty

……….

**Comment by martinhannigan** [comment]
And perhaps we need a “type 2” feature at some point which may make this
more palatable for others? Food for thought.

Thanks!

**Comment by martzuk** [comment]
@martinhannigan I agree with what you're saying, but I think the issue at hand is that the facility owner can decide to remove a network from their listing for any reason, whether that's by contract or otherwise, PeeringDB doesn't know.

I can think of some examples from the past where a DC didn't want to allow competing IXPs to be present in their facility at all, so said IXP used a third party for reach into the site. Whether that's considered "remote" or not is a different thing, the fact of it is that IXP is directly reachable from that site. In this case the fac object owner would be able to prevent it from being displayed.

PeeringDB is a data source, and IMO it should ultimately end with the network/IXP to state where they're present.

**Comment by grizz** [comment]
@martzuk You make a very compelling argument, I'm rethinking my position :)

**Comment by mustangthz** [comment]
I've asked the admin committee to remove 3 IXP's linked to my facilities.  They were all tunnel IXPs linking to sites to appear more real.   The bottom line is that both parties, facility owners and IXPs/Networks/Carriers, need the ability to remove people from the facility record.  It's not fair to the facility owners that we can't manage our facilities in the database.

**Comment by martzuk** [comment]
Right, but doing it via the AC IMO is the right workflow, rather than giving the facility owner unchecked power to decide who is allowed and who isn't.

**Comment by mustangthz** [comment]
It's not unchecked power, it's parity with the networks.  If a network is present at a facility and the owner is removing them, they should open a ticket with the AC.

**Comment by martzuk** [comment]
I agree, if a facility owner wants a network removed they should open a ticket with the AC.

**Comment by mustangthz** [comment]
No, it's not necessary to drive additional work to the AC.  The facility owners should have the ability to remove IX's, Networks, and Carriers.

**Comment by martzuk** [comment]
> @martinhannigan I agree with what you're saying, but I think the issue at hand is that the facility owner can decide to remove a network from their listing for any reason, whether that's by contract or otherwise, PeeringDB doesn't know.
> 
> I can think of some examples from the past where a DC didn't want to allow competing IXPs to be present in their facility at all, so said IXP used a third party for reach into the site. Whether that's considered "remote" or not is a different thing, the fact of it is that IXP is directly reachable from that site. In this case the fac object owner would be able to prevent it from being displayed.
> 
> PeeringDB is a data source, and IMO it should ultimately end with the network/IXP to state where they're present.

How would you mitigate the situation I described here?

**Comment by mustangthz** [comment]
By your own admission, in the scenario above, I would open a ticket with the AC and request they remove the IX from my facility.  They would remove it, and that would be the end of it.  You can't get around that the facility owner fundamentally owns their record.

**Comment by grizz** [comment]
> By your own admission, in the scenario above, I would open a ticket with the AC and request they remove the IX from my facility. They would remove it, and that would be the end of it. You can't get around that the facility owner fundamentally owns their record.

That would be a policy change then, not a feature.

*backs away from this issue* :)

**Comment by arnoldnipper** [comment]
> I agree, if a facility owner wants a network removed they should open a ticket with the AC.

@peeringdb/ac is **not** in the position to decide whether a carrier/ix/net is present at a facility. Hence, let's trust the facility owner. A button also could 

- send a notification to the {carrier,ix,net} admins, the facility admins and AC
- remove the objects after x working days (so far I don't see a need for immediate removal)

May I also suggest that we ammend the [Data Ownership Policy Document](https://docs.peeringdb.com/gov/misc/2020-04-06_PeeringDB_Data_Ownership_Policy_Document_v1.0.pdf)

**Comment by peterhelmenstine** [comment]
Even as a facility owner I still don't really know who's in my facility. In a telco hotel where people are a customer of a customer of a customer it's nearly impossible to be sure. $0.02

**Comment by martinhannigan** [comment]
True. However , your leases, MSA’s, etc and rules and regulations probably prohibit that conduct without your “reasonable”’consent.

If there is a benefit to the DC or they are not  contravening the DC’s policy they wont get clipped. That decision should never be up to PDB.  Its directory. Not a tenant advocacy service.

$0.02



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by peterhelmenstine** [comment]
I guess the landlord would have the right to control who's in their property.

**Comment by martinhannigan** [comment]
We should talk more about what it looks like when you want to de-link a network.



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by arnoldnipper** [comment]
> We should talk more about what it looks like when you want to de-link a network.

... or a carrier or an ix. So, why don't we develop a policy along what we did with the `netixlan` object in conjunction with the IX-F import? Let's give facility owners and ix operators a button to flag an {carrier,ix,net} object for removal. This then will kick off the same procedure as we do today for the `netixlan` object.

**Comment by martinhannigan** [comment]
+1

**Comment by mcmanuss8** [comment]
Let's do this: 

When a facility removes an object, an email is sent to the object being remove that lets them know they are being removed by the facility and gives them a clear path to escalate if they feel it is incorrect/malicious: email support@ to dispute.

**Comment by mustangthz** [comment]
+1On Jul 6, 2023, at 12:30 PM, Peter Helmenstine ***@***.***> wrote:﻿
+1

**Comment by grizz** [comment]
> When a facility removes an object, an email is sent to the object being remove that lets them know they are being removed by the facility and gives them a clear path to escalate if they feel it is incorrect/malicious: email support@ to dispute.

+1

**Comment by arnoldnipper** [comment]
Any reason we don't have it in line with what we do with the `netixlan` object?

**Comment by peterhelmenstine** [comment]
I always thought the IX had the ability to remove a LAN but I was corrected today on the product call. I was able to remove LANs because of my admin privileges so I guess I would like to know how it works as an IX operator. Should be similar in my opinion.

**Comment by arnoldnipper** [comment]
For IX operators we have to differentiate

1. if you provide an IX-F JSON, we follow the recommendations from the [DOTF](https://docs.peeringdb.com/taskforce/dataownership/). The missing part still is #1360 
2. if you don't provide an IX-F JSON, you have to mail @peeringdb/ac 

If we add a feature for removing objects (either for `fac` or `ix` operators), removal should be in line with 1., IMO

**Comment by leovegoda** [comment]
**Summary**

1. The owner of a `fac` object may request that @peeringdb/ac remove a `net`, `carrier`, or `ix` object connected to their `fac`
2. When the record is removed its owner will be notified of the removal
3. The owner of the removed object can appeal the removal to the AC and escalate to Stewards and Board decision

**Comment by martinhannigan** [comment]
I know I +1. I take it back.  :)  For a minute.

How can PDB override such a thing? Its the FAC’s data?

Thanks!

**Comment by arnoldnipper** [comment]
> How can PDB override such a thing? Its the FAC’s data?

We are in the same situation as with the `netixlan` object. IMO, we should apply the same [policy and procedures](https://docs.peeringdb.com/taskforce/dataownership/)

**Comment by mustangthz** [comment]
The FAC isn’t a multi-party data element.

The rules do not apply.

-Chris

**Comment by arnoldnipper** [comment]
> The FAC isn’t a multi-party data element.

That's true. However, we are talking about `carrierfac`, `ixfac` and `netfac`. All of them being multi-party data elements

**Comment by mustangthz** [comment]
None of them have anything to do with geocode for the FAC object.

**Comment by arnoldnipper** [comment]
> None of them have anything to do with geocode for the FAC object.

That's also true. However, we are not talking about geocode for the FAC object.

**Comment by mustangthz** [comment]
Doh!

I think FAC’s should be able to delete anything they want off their object.

-Chris

**Comment by martinhannigan** [comment]
Agreed.

**Comment by martinhannigan** [comment]
To be clear, what I mean is that there should be no appeal process. If it’s
on the fax data, they should be able to delete it. Without any arguments or
drawn out appeals from folks that really don’t have any interest in the
relationship.

**Comment by job** [comment]
-1 from me

I do not believe facility owners *in all cases* are aware who the indirect constituents are in a given facility.

Facility X rents a few racks worth of space to direct customer X, direct customer X sublets (physically, virtually, or optically) some resources to party Y; party Y lists themselves in PeeringDB as present in Facility X.

How is it in either Facility X's or customer X's or dependent Y's best interest when Facility X can unilaterally remove Y?

Am I missing something? We do have a handful of examples in the industry where Colocation Facilities somehow engaged in exclusive contracts with IX's, following that pattern to its logical conclusion the colo would need to delete all competing IX's in that facility that were reachable via a non-intuitive intermediate.

**Comment by mustangthz** [comment]
To your point, if a FAC owner mailed the AC an asked them to remove an IX from their FAC object, I believe they would do it.

What’s the difference between the AC doing it and the FAC owner being able to do it themselves?

**Comment by job** [comment]
To further the discussion let me introduce a tangible example: there are gazillion fibers between NIKHEF and other facilities in a radius of 300 meters [source](https://barm.nikhef.nl/housing/connectivity/).

If an IX or Network entity claims to have presence in NIKHEF, their actual presence might be an ODF panel in NIKHEF, or part of an ODF panel. NIKHEF has no means to verify which entity exactly is at the other end of the *short* optical connection.

Should NIKHEF be allowed to delist an IX or Network because that IX or Network isn't on the 'direct customer' list? I am not sure that is beneficial to the purpose of PeeringDB.

Please help me understand what exactly the objective is here

**Comment by job** [comment]
Ultimately the IX and Network entities exist in the digital real, while Facility owners exist only in the realm of providing physical space, power, and physical interconnection. Anything that happens through the optical fibers (in the digital realm) is outside the scope of the Facility owner: they have no way of proving to anyone that a given fiber optic doesn't somehow connect to a given IX or Network. Facilities having unilateral say and there being no appeals process for IX 
& Network entities seems destructive and to Marty's [point](https://github.com/peeringdb/peeringdb/issues/1366#issuecomment-1508253146) really disruptive to how this industry conducts business.

**Comment by peterhelmenstine** [comment]
I agree that a facility owner will never be able to confirm who's in their own building. I believe there was a concern that the landlord should have the ability to market the facility publicly based on their contracted customers and may not some listed for whatever reason.

Current policy requires the network to list themselves in a facility. Changing this would then put the onus back on the facility to list those networks which could have the opposite effect and make the data worse.

I'm changing my +1 to abstain.

**Comment by mustangthz** [comment]
I didn’t say an appeal process didn’t exist.

They likely have one.

Your construct is complete nonsense.  That a facility owner is some how a second class citizen of PDB is ridiculous.  Take your apartheid somewhere else.

-Chris

**Comment by job** [comment]
> I didn’t say an appeal process didn’t exist. They likely have one. Your construct is complete nonsense. That a facility owner is some how a second class citizen of PDB is ridiculous. Take your apartheid somewhere else.

wtf?

**Comment by mustangthz** [comment]
If you couldn’t tell I was kidding. There’s some nuance to work though on this. On Nov 2, 2023, at 8:16 PM, Job Snijders ***@***.***> wrote:﻿

I didn’t say an appeal process didn’t exist. They likely have one. Your construct is complete nonsense. That a facility owner is some how a second class citizen of PDB is ridiculous. Take your apartheid somewhere else.

wtf?

**Comment by mustangthz** [comment]
Sounds good to meOn Nov 2, 2023, at 8:28 PM, Matt Griswold ***@***.***> wrote:﻿

When a facility removes an object, an email is sent to the object being remove that lets them know they are being removed by the facility and gives them a clear path to escalate if they feel it is incorrect/malicious: email support@ to dispute.

To be clear, I don't have much of an opinion on this either way, but I will say this was a compromise from all parties on the multiple calls the PC devoted to it, it passed with consensus, and is slated to be started on this month.
As I recall, the thought process was that carriers are starting to put a lot of entries into PDB as a sort of means for advertising, and we allow that to happen without facility approval, we should also allow facilities to also remove them. We discussed the possibility of facilities doing this maliciously, figured it wasn't likely to happen, and all agreed that with a clear reporting path we could deal with abuse on a case by case basis.

**Comment by arnoldnipper** [comment]
> Ultimately the IX and Network entities exist in the digital realm, while Facility owners exist only in the realm of providing physical space, power, and physical interconnection. Anything that happens through the optical fibers (in the digital realm) is outside the scope of the Facility owner: they have no way of proving to anyone that a given fiber optic doesn't somehow connect to a given IX or Network. Facilities having unilateral say and there being no appeals process for IX
> & Network entities seems destructive and to Marty's [point](https://github.com/peeringdb/peeringdb/issues/1366#issuecomment-1508253146) really disruptive to how this industry conducts business.

+1 what you say, @job. Wouldn't adoption of our `netixlan` DOTF policy and procedure solve the issue?  The `fac` operator may request deletion of an `carrierfac`, `ixfac`, or `netfac` object. But w/o ACK from the other side or timeout, the object would not be deleted.

The framework is there. Both document- and implementation wise.

**Comment by martinhannigan** [comment]
On Thu, Nov 2, 2023 at 19:56 Job Snijders ***@***.***> wrote:

> -1 from me
>
> I do not believe facility owners *in all cases* are aware who the
> indirect constituents are in a given facility.
>
> Facility X rents a few racks worth of space to direct customer X, direct
> customer X sublets (physically, virtually, or optically) some resources to
> party Y; party Y lists themselves in PeeringDB as present in Facility X.
>

Probably violating a customer agreement which PDB has no role in if they
do. Otherwise, they're aligned and no need.  Commercial interests and
contracts will manage behavoirs.

How is it in either Facility X's or customer X's or dependent Y's best
> interest when Facility X can unilaterally remove Y?
>

Thats right.  But thats between all of them. Not us.


Am I missing something? We do have a handful of examples in the industry
> where Colocation Facilities somehow engaged in exclusive contracts with
> IX's, following that pattern to its logical conclusion the colo would need
> to delete all competing IX's in that facility that were reachable via a
> non-intuitive intermediate.
>

The only reason Id delete an entry is if they were violating customer
agreements or acting in an unaligned way -and not a legit customer. If
they’re legit in the facility any issue is going to be a corner case which
we wont be able to define here. Contracts determine who is legit. Courts
mediate disputes. Interfering is not our role.

And then data ownership policy. And accuracy.

Warm regards,

-M<

**Comment by martinhannigan** [comment]
On Thu, Nov 2, 2023 at 20:37 Arnold Nipper ***@***.***> wrote:

> Ultimately the IX and Network entities exist in the digital realm, while
> Facility owners exist only in the realm of providing physical space, power,
> and physical interconnection. Anything that happens through the optical
> fibers (in the digital realm) is outside the scope of the Facility owner:
> they have no way of proving to anyone that a given fiber optic doesn't
> somehow connect to a given IX or Network. Facilities having unilateral say
> and there being no appeals process for IX
> & Network entities seems destructive and to Marty's point
> <https://github.com/peeringdb/peeringdb/issues/1366#issuecomment-1508253146>
> really disruptive to how this industry conducts business.
>
> +1 what you say, @job <https://github.com/job>. Wouldn't adoption of our
> netixlan DOTF policy and procedure solve the issue? The fac operator may
> request deletion of an carrierfac, ixfac, or netfac object. But w/o ACK
> from the other side or timeout, the object would not be deleted.
>
> The framework is there. Both document- and implementation wise.
>

An IXP (or other) has no rights to circumvent the owner of a facility (or
even be in the facility physically) using a proxy in violation of (sublet
clauses etc) policies and agreements real or verbal and should never be
facilitated by PDB in doing so. The IXP has remedies in the civil system.
If PDB interferes I see much liability. If one owns the fac and object it
seems reasonable its up to them - per our own policy.

-1 with appeals
+1 without

We’re not breaking new ground here. This is IRL.


Thanks

-M<

**Comment by nbakker** [comment]
This discussion seems facility-dominated. I understand the field to mean "I'm ready to arrange for a PNI somewhere in this facility via some means", not an endorsement of the existence of a contractual relationship.

Are there any cases where a facility operator was questioned about access to a carrier or network at a facility because the carrier or network made an incorrect listing? Should PeeringDB add a heading on a facility's page to indicate that the IXP and carrier lists are supplied by those and not the facility?

Perhaps we can revert to community management for facilities, that way we can reverse horrorshow entries like Interxion/DRT MRS1,2,3 which are two different campuses but grouped into one entry.

**Comment by eimann** [comment]
In my routine work with various types of facilities, I’ve noticed a concerning issue I’d like to point out. Allowing facilities to delete records at will gives them too much control, a level of control they don’t always manage properly in real life. There have been several situations where facilities couldn't understand their own contracts without a lawyer’s help, even when the terms were clear. There's also a worrying increase in facilities trying to overstep their bounds with direct tenants.

The issue becomes more serious with PeeringDB. When a facility user can change a facility's information in PeeringDB, it creates a mismatch between what facilities think is real and the actual reality for operators. This leads to a trust issue with the data we use to run our operations.

**Comment by martinhannigan** [comment]
Declaring an emergency conference call regarding this issue.

We're all on the same page but we just don't know it. Email is the WORST
discussion medium ever.

Best timing possible. It's OK if you can't make it. Guaranteed someone will
make your point based on the github discussion.


*Reply directly for a direct invite for your calendar:*



*Join on your computer, mobile app or room device*

Click here to join the meeting
<https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZGUyMDg4MmEtMzU3Ny00OTk4LWI0NGUtMTFlYzQyOWRmODM3%40thread.v2/0?context=%7b%22Tid%22%3a%227da18d44-7ba0-4051-89de-c230b69439d2%22%2c%22Oid%22%3a%22796db1c3-65f8-4c75-b78b-0590a1f3c96e%22%7d>

Meeting ID: 233 747 289 87
Passcode: rfhDjx

**Comment by peterhelmenstine** [comment]
I equate this to a landlord controls the directory board when you walk into a lobby but they don't always know who's in the building. Google Maps is in a similar scenario. Curious what their policy is when someone claims a business a building.

**Comment by martinhannigan** [comment]
Github updated and sent here as well with specific

Wednesday, November 8, 2023 from* 10:30 AM to 11:30 AM (EDT BOSTON, USA)
or 2:30PM UTC*

*Join on your computer, mobile app or room device*

Click here to join the meeting
<https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZGUyMDg4MmEtMzU3Ny00OTk4LWI0NGUtMTFlYzQyOWRmODM3%40thread.v2/0?context=%7b%22Tid%22%3a%227da18d44-7ba0-4051-89de-c230b69439d2%22%2c%22Oid%22%3a%22796db1c3-65f8-4c75-b78b-0590a1f3c96e%22%7d>

Meeting ID: 233 747 289 87
Passcode: rfhDjx

**Comment by martinhannigan** [comment]
On Fri, Nov 3, 2023 at 1:33 PM Dominik Bay ***@***.***> wrote:

> In my routine work with various types of facilities, I’ve noticed a
> concerning issue I’d like to point out. Allowing facilities to delete
> records at will gives them too much control, a level of control they don’t
> always manage properly in real life. There have been several situations
> where facilities couldn't understand their own contracts without a lawyer’s
> help, even when the terms were clear. There's also a worrying increase in
> facilities trying to overstep their bounds with direct tenants.
>

IXPs over step their boundaries all the time. So do network operators. 'm
not sure what the difference is? No one should tell an IXP who they should
list as their member and no one should tell an operator who they must have
as a customer. Why is a building owner or facility operator different?

The amount of deletion likely to happen is low and related to hygiene and
abuse by bad actors. However, the point is about appeal. PDB is not the
police, has no such powers and shouldn't put its fingers on any commercial
issues between two parties. If those parties have a legitimate relationship
they will work it out fine without PDB. Or not.

The issue becomes more serious with PeeringDB. When a facility user can
> change a facility's information in PeeringDB, it creates a mismatch between
> what facilities think is real and the actual reality for operators. This
> leads to a trust issue with the data we use to run our operations.
>

Got a live example?

**Comment by martinhannigan** [comment]
On Thu, Nov 2, 2023 at 8:07 PM Job Snijders ***@***.***>
wrote:

> To further the discussion let me introduce a tangible example: there are
> gazillion fibers between NIKHEF and other facilities in a radius of 300
> meters source <https://barm.nikhef.nl/housing/connectivity/>.
>
> If an IX or Network entity claims to have presence in NIKHEF, their actual
> presence might be an ODF panel in NIKHEF, or part of an ODF panel. NIKHEF
> has no means to verify which entity exactly is at the other end of the
> *short* optical connection.
>
> Should NIKHEF be allowed to delist an IX or Network because that IX or
> Network isn't on the 'direct customer' list? I am not sure that is
> beneficial to the purpose of PeeringDB.
>

This is the wrong way to think about it. If NIKHEF cares, then yes.
Ultimately, it's their building and network. They set the policies. No one
has a "right" to be at NIKHEF. They have a contractual obligation or a
benefit. I can't imagine what someone is going to appeal if they ever do if
a data center wants them cleaned off of the site.

**Comment by martinhannigan** [comment]
I thought about this over the weekend. For some clarity:

On Thu, Nov 2, 2023 at 7:56 PM Job Snijders ***@***.***>
wrote:

> -1 from me
>
> I do not believe facility owners *in all cases* are aware who the
> indirect constituents are in a given facility.
>
You're talking about proxies. They're not third party constituents, they're
customers. If their host, the proxy, has an agreement with the
owner/operator to engage in business this way there's no problem. In fact,
this happens all the time without any issue or need for anyone to get into
the middle of it.

Facility X rents a few racks worth of space to direct customer X, direct
> customer X sublets (physically, virtually, or optically) some resources to
> party Y; party Y lists themselves in PeeringDB as present in Facility X.
>
> How is it in either Facility X's or customer X's or dependent Y's best
> interest when Facility X can unilaterally remove Y?
>
If they remove the legitimate PDB listing of a customer in their data
center they are depriving themselves of potential income "alignment". It
seems like we're talking about the most extreme corner case possible
otherwise.

Am I missing something? We do have a handful of examples in the industry
> where Colocation Facilities somehow engaged in exclusive contracts with
> IX's, following that pattern to its logical conclusion the colo would need
> to delete all competing IX's in that facility that were reachable via a
> non-intuitive intermed
>
Any current examples of this? I'm not aware of any. The solution seems to
involve obtaining a valid presence in a facility or  data center, either
through a legit proxy or directly. It's unacceptable for PDB to be
complicit in breaching any entity's policies.

This looks like pretty healthy ecosystem with good alignment owner/operator
facility, DC and IXP:

https://www.peeringdb.com/ix/804
https://www.peeringdb.com/ix/944
https://www.peeringdb.com/ix/777
https://www.peeringdb.com/ix/155
https://www.peeringdb.com/ix/1679

Self-service benefits everyone. Ideally, we should prioritize self-service
end-to-end to enhance user convenience, reduce operational costs (often a
concern among committees), and ensure accuracy. While occasional exceptions
might exist among numerous entries, favoring manual intervention poses
antitrust risks and increases liability, potentially compromising
indemnification. Putting fingers on the scale is not an activity we want to
engage in. It's a choice between embracing self-service entirely or not at
all. Relying on appeals isn't practical and tends to generate more issues
than solutions.

Hope that helps.

Warm regards,

-M<

**Comment by martzuk** [comment]
Not to throw too much fuel to the fire, but it seems like those "associated" with a facility are the ones advocating for unchecked power, whereas other want a more nuanced approach. Since it was mentioned facility owners aren't likely to want to delete super often, it would likely cause very little burden on the AC. One argument I don't buy however is "I'd never maliciously delete anybody from my facility, therefore it's not an issue". Whether you would or not, it doesn't stop somebody from doing it maliciously. Having an appeals process just provides a mechanism to undo said malice.

**Comment by martinhannigan** [comment]
On Mon, Nov 6, 2023 at 10:39 AM Marty Strong ***@***.***>
wrote:

> Not to throw too much fuel to the fire, but it seems like those
> "associated" with a facility are the ones advocating for unchecked power,
> whereas other want a more nuanced approach. Since it was mentioned facility
> owners aren't likely to want to delete super often, it would likely cause
> very little burden on the AC. One argument I don't buy however is "I'd
> never maliciously delete anybody from my facility, therefore it's not an
> issue". Whether you would or not, it doesn't stop somebody from doing it
> maliciously. Having an appeals process just provides a mechanism to undo
> said malice.
>

Appreciate that. This is analogous to saying we shouldn't peer with a
network using IPv5. Haven't seen an example of how deletion would be
malicious and more than an issue between the two parties. Or possibly three
if a proxy were involved. The reason I'm adamant on the appeal is they
don't work, there isn't a process already and its a precedent to try and
extend this to other things that probably don't need appeals as well.
Imagine someone "appealing" a geo code edit? Similar concept.

Thinking about how to move this ahead somehow, there isn't an appeals
process here. There is a sort of wishy-washy exception process
https://docs.peeringdb.com/committee/admin/approval-guidelines/#exception-process
which says a requestor may ask for "more review". If we follow that and we
do not restore an entry to the very end of the process (and codify it in
the issue) that might work since I would argue there will be no
circumstance that is going to result in a deletion being undone. Like IPv5.

Warm regards,

-M<

**Comment by arnoldnipper** [comment]
> Having an appeals process just provides a mechanism to undo said malice.

Applying the `netixlan` process and procedures would be an easy solution to the issue, IMO. Everything is in place and we don't have to reinvent the wheel.

**Comment by nbakker** [comment]
The 'netixlan' process was put in place because IP addresses are unique identifiers (and are constrained as such in the database).
Declarations of presence at facilities aren't that.

**Comment by arnoldnipper** [comment]
> The 'netixlan' process was put in place because IP addresses are unique identifiers (and are constrained as such in the database).
> Declarations of presence at facilities aren't that.

The [DOTF](https://docs.peeringdb.com/taskforce/dataownership/) "... was established in September 2019 with the aim of working on a PeeringDB Policy proposal about data ownership, after a need was recognized by the Product Committee as issues consistently had been raised relating to who owns the data in PeeringDB when more than one party is involved (as in netixlan, ixfac, netfac, etc objects)."

We are done with the `netixlan`, but never made it to `carrierfac`, `ixfac`, and `netfac` we are now talking about.

**Comment by leovegoda** [comment]
The board discussed the DOFT outcomes and this issue at its January 2025 meeting. The outcome in the [published draft minutes](https://docs.peeringdb.com/gov/legaldocs/2025-01-23_PeeringDB_Board_Minutes.pdf): "Users remain in control of their data and in cases of disputes they can go to the Admin Committee for arbitration."

So I'll set the status back to Decide and won't schedule this work until there's an updated design.

**Comment by leovegoda** [comment]
Closing as there's been no comment for six month
